File: skills/code-project-analyze/tests/sample_project/src/api/wordCards.js
// src/api/wordCards.js —— 词卡 API 客户端（样例源码）
import axios from 'axios';

const BASE = import.meta.env.VITE_API_BASE || '/api/v4';

// 获取分组树
export async function getGroupTree() {
  return axios.get(`${BASE}/word-cards/groups/tree`).then(r => r.data);
}

// 创建词卡
export async function createCard(payload) {
  return axios.post(`${BASE}/word-cards`, payload).then(r => r.data);
}

// 更新词卡
export async function updateCard(id, payload) {
  return axios.put(`${BASE}/word-cards/${id}`, payload).then(r => r.data);
}

// 删除词卡
export async function deleteCard(id) {
  return axios.delete(`${BASE}/word-cards/${id}`).then(r => r.data);
}

// 语义搜索
export async function searchCards(query) {
  return axios.post(`${BASE}/word-cards/search`, { q: query }).then(r => r.data);
}

File: skills/code-project-analyze/tests/sample_project/backend/api/word_cards.py
# backend/api/word_cards.py —— 词卡后端 API（样例源码 / FastAPI）
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel

router = APIRouter(prefix="/api/v4/word-cards", tags=["word_cards"])


class WordCardIn(BaseModel):
    title: str
    content: str
    group_id: int | None = None


@router.get("/groups/tree")
async def get_group_tree():
    """返回分组树形结构"""
    return {"roots": []}


@router.post("")
async def create_card(payload: WordCardIn):
    """创建词卡"""
    return {"id": 1, **payload.model_dump()}


@router.put("/{card_id}")
async def update_card(card_id: int, payload: WordCardIn):
    """更新词卡"""
    if card_id <= 0:
        raise HTTPException(status_code=404, detail="not found")
    return {"id": card_id, **payload.model_dump()}


@router.post("/search")
async def search_cards(body: dict):
    """语义搜索词卡"""
    return {"results": [], "query": body.get("q", "")}
